orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/HttpClientUtils.java

```diff
@@ -160,9 +175,44 @@ public String httpGetAsString(String url) throws IOException {
     }
   }
 
+  private int getProxyPort() {
+    String proxyPort = System.getProperty(JVM_HTTP_PROXY_PORT);
+    int defaultProxyPort = 8080;
+
+    if (proxyPort != null) {
+      try {
+        return Integer.parseInt(proxyPort);
+      } catch (NumberFormatException e) {
+        LOGGER.warn(
+            "Invalid proxy port number: {}. Using default port: {}",
+            proxyPort,
+            defaultProxyPort);
+      }
+    }
+    return defaultProxyPort;
+  }
+
   class RetryRequestException extends RuntimeException {
     RetryRequestException(String cause) {
       super(cause);
     }
   }
+
+  class CustomRedirectStrategy {
+    public RedirectStrategy getLaxRedirectStrategy() {
+      return new LaxRedirectStrategy() {
+        @Override
+        public boolean isRedirected(
+            HttpRequest request, HttpResponse response, HttpContext context) {
+
+          int statusCode = response.getStatusLine().getStatusCode();
+          if ((statusCode >= 300) && (statusCode <= 399)) {
```

Comment:
I'd not run the full range, as each status code is supposed to be handled SOMEWHAT differently.
* 301 & 302 - follow & go to the requested URL using "GET" method
* 307 & 308 - this one is trickier.  This is "make a request using the SAME requesting method and PAYLOAD to the new URL".  NORMALLY the requirement is ALSO to prompt the user to ENSURE they really want this.  I'd be hesitant given the situation with spinnaker to support 307/308 redirects in any shape, and ONLY support 301/302s (which should cover MOST use cases)

Reply:
https://www.rfc-editor.org/rfc/rfc9110.html#name-redirection-3xx FYI.  `Automatic redirection needs to be done with care for methods not known to be [safe](https://www.rfc-editor.org/rfc/rfc9110.html#safe.methods), as defined in [Section 9.2.1](https://www.rfc-editor.org/rfc/rfc9110.html#safe.methods), since the user might not wish to redirect an unsafe request.` PARTICULARLY given this can expose URLs that are NOT safe... there's definitely a "caution" flag here.  E.g. you get into link local redirection follows for pulling metadata WHICH SHOULD NOT be allowed.  


ANOTHER THING NEEDED:
UserConfiguredUrlRestrictions.validateURI on the redirect URL MUST BE CALLED.  Or you're opening up link local attacks.  Since I think this is recreating the http client as part of retry operations...